WordPressVIPMinimum/Sniffs/Variables/ServerVariablesSniff.php

```diff
@@ -56,14 +59,22 @@ public function process( File $phpcsFile, $stackPtr ) {
 		}
 
 		$variableNamePtr = $phpcsFile->findNext( array( T_CONSTANT_ENCAPSED_STRING ), ( $stackPtr + 1 ), null, false, null, true );
-		$variableName    = str_replace( "'", '', $tokens[ $variableNamePtr ]['content'] );
+		$variableName    = str_replace( array( "'", '"' ), '', $tokens[ $variableNamePtr ]['content'] );
 
 		if ( false === in_array( $variableName, $this->restrictedVariables, true ) ) {
 			// Not the variable we are looking for.
 			return;
 		}
 
-		$phpcsFile->addError( 'Basic authentication should not be handled via PHP code.', $stackPtr, 'ServerVariables' );
+		if ( 'PHP_AUTH_PW' === $variableName ) {
+			$phpcsFile->addError( 'Basic authentication should not be handled via PHP code.', $stackPtr, 'ServerVariables' );
+		} elseif ( 'HTTP_X_IP_TRAIL' === $variableName || 'HTTP_X_FORWARDED_FOR' === $variableName || 'REMOTE_ADDR' === $variableName ) {
```

Comment:
Rather than comparing `=== $variableName` three times (and potentially other strings in the future), why not add the current three items to an array (use associative array with strings as keys and `true` as value - see [here](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/blob/1bc614137d5ed913ab2df00208b1edf5279dcb8e/WordPress/Sniff.php#L25-L34) for reason), and then use `isset()`?

```php
/**
 * {Add docs here as to why these are special}
 *
 * @since ...
 */
private $user_controlled_variables = array(
    'HTTP_X_IP_TRAIL'      => true,
    'HTTP_X_FORWARDED_FOR' => true,
    'REMOTE_ADDR'          => true,
);

...

} elseif ( isset( $this->user_controlled_variables[ $variableName ] ) {
```

The data (property) is then separate from the logic (method), the data can be documented (property DocBlock), and the logic is more self-documenting (property name).